uaclient/cli/formatter.py

```diff
@@ -0,0 +1,175 @@
+    show_suggestions = True
+
+    # Initializing the class after the import is useful for unit testing
+    @classmethod
+    def init(cls, cfg: UAConfig):
+        cls.use_utf8 = (
+            sys.stdout.encoding is not None
+            and "UTF-8" in sys.stdout.encoding.upper()
+        )
+
+        cls.use_color = (
+            sys.stdout.isatty()
+            and os.getenv("NO_COLOR") is None
+            and cfg.cli_color
+        )
+
+        cls.show_suggestions = cfg.cli_suggestions
+
+    @classmethod
+    def disable_color(cls) -> None:
+        cls.use_color = False
+
+    @classmethod
+    def disable_suggestions(cls) -> None:
+        cls.show_suggestions = False
+
+
+ProOutputFormatterConfig.init(cfg=UAConfig())
+
+
+def len_no_color(text: str) -> int:
+    return len(re.sub(COLOR_FORMATTING_PATTERN, "", text))
+
+
+# We can't rely on textwrap because of the len_no_color function
+# Textwrap is using a magic regex instead
+def wrap_text(text: str, max_width: int) -> List[str]:
+    if len_no_color(text) < max_width:
+        return [text]
+
+    words = text.split()
+    wrapped_lines = []
+    current_line = ""
+
+    for word in words:
+        if len_no_color(current_line) + len_no_color(word) >= max_width:
+            wrapped_lines.append(current_line.strip())
+            current_line = word
+        else:
+            current_line += " " + word
+
+    if current_line:
+        wrapped_lines.append(current_line.strip())
+
+    return wrapped_lines
+
+
+class Table:
+    SEPARATOR = " " * 2
+
+    def __init__(
+        self,
+        headers: Optional[List[str]] = None,
+        rows: Optional[List[List[str]]] = None,
+        max_length: Optional[int] = None,
+    ):
+        self.headers = headers if headers is not None else []
+        self.rows = rows if rows is not None else []
+        _all_content = self._validate_content()
+        if sys.stdout.isatty():
+            self.max_length = (
+                os.get_terminal_size().columns
+                if max_length is None
+                else max_length
+            )
+        else:
+            # If you're not in a tty, we don't care about wrapping
+            # If you have a thousand characters line on the table, well, wow
+            self.max_length = 999
+        column_sizes = []
+        for i in range(len(_all_content[0])):
+            column_sizes.append(
+                max(len_no_color(str(item[i])) for item in _all_content)
+            )
+        self.column_sizes = column_sizes
+
+    @staticmethod
+    def ljust(string: str, total_length: int) -> str:
+        str_length = len_no_color(string)
+        if str_length >= total_length:
+            return string
+        return string + " " * (total_length - str_length)
+
+    def _validate_content(self) -> List[List[str]]:
+        if not self.headers and not self.rows:
+            raise ValueError(
+                "Empty table not supported. Please provide headers or rows."
```

Comment:
I think we need to move those exception messages to the `messages` module

Reply:
The reason I didn't put those there is that this should not be translated.These tables should never depend on user input, and the network input that lands here needs to be sanitized in code.
Those issues are more like warnings to us, developers, to build the stuff the correct way.

If you disagree, we can address it in a separate PR